hibernate-reactive-core/src/test/java/org/hibernate/reactive/UpdateOrderingTestBase.java

```diff
@@ -0,0 +1,307 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.reactive.provider.Settings;
+import org.hibernate.reactive.testing.DatabaseSelectionRule;
+import org.hibernate.reactive.testing.SqlStatementTracker;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.unit.TestContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.POSTGRESQL;
+
+public abstract class UpdateOrderingTestBase extends BaseReactiveTest {
+
+	@Rule
+	public DatabaseSelectionRule rule = DatabaseSelectionRule.runOnlyFor( POSTGRESQL );
+
+	public static class UpdateOrderingTrueTest extends UpdateOrderingTestBase {
+
+		@Override
+		public boolean isUpdateOrdering() {
+			return true;
+		}
+
+		@Override
+		public String getBatchSize() {
+			return "0";
+		}
+
+		@Override
+		public void assertLoggedQueries() {
+			assertThat( sqlTracker.getLoggedQueries() ).hasSize( 30 );
```

Comment:
I think we can reduce the number of queries. No need to run 30 of them.
We can test with just 6 queries for example.